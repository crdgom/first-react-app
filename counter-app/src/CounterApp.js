import PropTypes from 'prop-types';
import { useState } from 'react';

const CounterApp = ({name, value}) => {
    const [counter, setCounter]= useState(100);
    // handleAdd
    const handleAdd = ()=>{
        setCounter(counter +1);
        //setCounter( (c)=> c+1)
    }
    const handleRemove =()=>{
        setCounter(counter -1);
    }
    const handleReset =()=>{
        setCounter(value = 100)
    }
    return (
        <>
            <h1>{name}</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleRemove}>-1</button>
        </>
    );
}

CounterApp.defaultProps = {
    name: 'CounterApp',
    value: PropTypes.number
}
export default CounterApp