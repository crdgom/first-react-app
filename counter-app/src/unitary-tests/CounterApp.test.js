import React from 'react';
import { render } from '@testing-library/react';
import CounterApp from "../CounterApp";
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Testing for CounterApp component',()=>{

    test('Must show default value defined in CounterApp component',()=>{
        const value = 100;
        const wrapper = shallow(<CounterApp value={value}/>);
        expect(wrapper).toMatchSnapshot();
    })
    test('must show the h2 whith value 100', ()=>{
        const value = '100';
        const wrapper = shallow(
            <CounterApp
                value = {value}
            />)
        const heading2 = wrapper.find('h2').text() 
        expect(heading2).toBe(value);
    })

})

describe('Testing for CounterApp buttons (plus 1, reset & minus 1)',()=>{
    let wrapper = shallow(<CounterApp />);

    beforeEach(()=>{

        wrapper = shallow(<CounterApp />);
        
    })
    test('Should simulate the behavior of the button plus one (+1) ',()=>{
        
        wrapper.find('button').at(0).simulate('click');
        const heading2 = wrapper.find('h2').text()
        expect(heading2).toBe('101')

    })
    
    test('Should simulate the behavior of the button minus one (+1) ',()=>{
        
        wrapper.find('button').at(2).simulate('click');
        const heading2 = wrapper.find('h2').text()
        expect(heading2).toBe('99')

    })

    test('Should simulate the behavior of the button reset (Reset) ',()=>{
        
        const wrapper = shallow(<CounterApp value={100}/>);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const heading2 = wrapper.find('h2').text()
        expect(heading2).toBe('100')
        
    })
})
