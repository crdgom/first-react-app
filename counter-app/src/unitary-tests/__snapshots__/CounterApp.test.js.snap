// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for CounterApp component Must show default value defined in CounterApp component 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
